### ISSUE TITLE
Returning deleted entities breaks

### ISSUE THREAD

**Issue by brian-gates** [neo4j/neo4j#6274]
```
CREATE (a)-[r:related]->(b) 
DELETE r 
RETURN *
```

Results in `Neo.DatabaseError.Statement.ExecutionFailure` with message "Relationship not found"

I'd expect this not to fail, or at least specify _which_ relationship is unfound.

Slack context: https://neo4j-users.slack.com/archives/help-cypher/p1453409530000617

**Comment by Mats-SX** [comment]
Thanks for reporting this @brian-gates. We'll look at it immediately. The bug seems to be related to returning deleted entities. If you alter your query to 

```
CREATE (a)-[r:related]->(b) 
DELETE r 
RETURN a, b
```

you should be rid of the bug.

**Comment by Mats-SX** [comment]
This behaviour will not be changed in 2.3, but #6354 should solve this, and the solution will be available in `3.0.0-M04`. Please reopen this issue if it persists.
